### DISCUSSION TITLE
PC fan control [issue]

### DISCUSSION THREAD

**Discussion by Bimbovadasz** [supermerill/SuperSlicer#1253] [Q&A]
Hello there!

I'm trying to make custom g-codes for a tool changer build and I want to control the part cooling fans. I found variables like min_fan_speed and max_fan_speed, but I want to use the value that SuperSlicer calculated not what I set, and I can't find a variable for that. Is there some way to access that value?

Thanks!

**Comment by supermerill** [comment]
no, it isn't, as it's currently computed after the custom is injected.

**Reply by Bimbovadasz** [reply]
Oh I see, that's unfortunate. Thank you for the answer!
 
I also have problems with the slicer putting M106 in the final gcode where I don't want to and I didn't specified to. 
In my TC gcode in line 25 is the tool change for the next extruder, and the slicer put an M106 before that. So in the final gcode the previous pc fan turns off then turn back on, because it is before the T[next_extruder] line. So the pc fan of the previous hot end remains on while the toolhead is inactive. I can't find a solution for this, but I might be doing something wrong.
I hope my description is not too confusing. 

My gcode: 
![TC gcode](https://user-images.githubusercontent.com/35198213/120892677-c19c7f00-c60f-11eb-83c8-2257d4d4dad5.PNG)
Final gcode:
![gcode with M106 inserted](https://user-images.githubusercontent.com/35198213/120892680-c3664280-c60f-11eb-8955-44172c4b6033.PNG)

**Reply by supermerill** [reply]
I don't know... I need some time to check that.

**Reply by Bimbovadasz** [reply]
Of course, thank you!

**Reply by supermerill** [reply]
I forgot this "issue as it's here instead of a proper issue.
If you want me to look at your discovered bug, please create a proper issue with project file, etc.

**Reply by supermerill** [reply]
#3874
